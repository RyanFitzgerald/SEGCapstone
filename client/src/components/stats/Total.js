import React from 'react';
import ChartsWrapper from '../ChartsWrapper';

class Total extends React.Component {
  constructor() {
    super();

    // Bind functions
    this.handleFilter = this.handleFilter.bind(this);
  }

  componentDidMount() {
    // Set title
    document.title = 'Total Sales Volume | Renovaction';

    // Update active tab
    this.props.setActiveSubtab(1);
  }

  handleFilter() {

  }

  render() {
    const data = {
      labels: ["January", "February", "March", "April", "May", "June"],
      datasets: [{
        label: 'Sales Volume',
        data: [12, 19, 3, 5, 2, 3],
        backgroundColor: [
          'rgba(255, 99, 132, 0.2)',
          'rgba(54, 162, 235, 0.2)',
          'rgba(255, 206, 86, 0.2)',
          'rgba(75, 192, 192, 0.2)',
          'rgba(153, 102, 255, 0.2)',
          'rgba(255, 159, 64, 0.2)'
        ],
        borderColor: [
          'rgba(255,99,132,1)',
          'rgba(54, 162, 235, 1)',
          'rgba(255, 206, 86, 1)',
          'rgba(75, 192, 192, 1)',
          'rgba(153, 102, 255, 1)',
          'rgba(255, 159, 64, 1)'
        ],
        borderWidth: 1
      }]
    };

    const options = {
      scales: {
        yAxes: [{
          ticks: {
            beginAtZero:true
          }
        }]
      }
    };

    return (
      <div className="content">
        <div className="row">
          <div className="column">
            <h2 className="card-title">Filter Data</h2>
            <div className="card">
              <div className="row">
                <div className="sm-4 column">
                  <label className="form-label" htmlFor="month">Month</label>
                  <span className="form-select">
                    <select ref={input => this.month = input} id="month" name="month" onChange={this.handleFilter}>
                      <option value="">All</option>
                      <option value="january">January</option>
                      <option value="february">February</option>
                      <option value="march">March</option>
                    </select>
                  </span>
                </div>
                <div className="sm-4 column">
                  <label className="form-label" htmlFor="year">Year</label>
                  <span className="form-select">
                    <select ref={input => this.month = input} id="year" name="year" onChange={this.handleFilter}>
                      <option value="">All</option>
                      <option value="2017">2017</option>
                      <option value="2016">2016</option>
                      <option value="2015">2015</option>
                    </select>
                  </span>
                </div>
                <div className="sm-4 column">
                  <label className="form-label" htmlFor="postalCode">Postal Code</label>
                  <input ref={input => this.postalCode = input} id="postalCode" name="postalCode" className="form-text" type="text" onKeyUp={this.handleFilter}/>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div className="row">
          <div className="column">
            <h2 className="card-title">Total Sales Volume</h2>
            <div className="card">
              <div className="row">
                <div className="md-4 column stats-number">
                  <span>$23,232.00</span>
                  <h3>Total Sales</h3>
                </div>
                <div className="md-8 column stats-chart">
                  <ChartsWrapper type="line" data={data} options={options}/>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Total;