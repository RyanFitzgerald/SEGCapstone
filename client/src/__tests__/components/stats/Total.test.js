import React from 'react';
import Total from '../../../Components/stats/Total';

describe('<Total />', () => {

  beforeAll(() => {
    // Add necessary info to user
    const user = {
      role: {
        level: 1
      }
    };
    localStorage.setItem('user', JSON.stringify(user));

  });

  it('should render correctly', () => {
    // Return empty array on requests
    mock.onAny().reply(200, []);
    // Create mock functions / spies
    const setActiveSubtabMock = jest.fn();
    const renderErrorMock = jest.fn();
    const getTotalVolumeSpy = jest.spyOn(Total.prototype, 'getTotalVolume');

    // Shallow render it
    const wrapper = shallow(
      <Total setActiveSubtab={setActiveSubtabMock} renderError={renderErrorMock}/>
    );

    // Compare to old snapshot
    expect(shallowToJson(wrapper)).toMatchSnapshot();

    // Check set active subtab is called and with correct value
    expect(getTotalVolumeSpy).toHaveBeenCalledTimes(1);
    expect(setActiveSubtabMock).toHaveBeenCalledTimes(1);
    expect(setActiveSubtabMock.mock.calls[0][0]).toBe(1);
  });
});
