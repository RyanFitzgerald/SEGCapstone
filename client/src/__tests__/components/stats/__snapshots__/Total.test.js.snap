// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Total /> should render correctly 1`] = `<Loading />`;
